<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Anna V. Protasio" />


<title>B cluster</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Bioinfo. analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Intro</a>
</li>
<li>
  <a href="All_clusters.html">All sequences</a>
</li>
<li>
  <a href="B_cluster.html">B cluster</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">B cluster</h1>
<h4 class="author">Anna V. Protasio</h4>
<h4 class="date">16/10/2019</h4>

</div>


<!-- Load libraries. -->
<p>In this section we outline the analyses done to investigate:</p>
<ol style="list-style-type: decimal">
<li>Frquencies of individual residues in VIF protein at positions 31, 33 and 128.</li>
<li>Frquencies of paires of residues at positions 31 and 128.</li>
<li>Phylogenetic distribution of proteins with different (and selected) residue combinations.</li>
</ol>
<p><strong>WE FOCUS ONLY ON CLUSTER B SEQUENCES</strong></p>
<div id="data-aquisition." class="section level1">
<h1>Data aquisition.</h1>
<p>The <strong>Los Alamos</strong> database was queried to obtain the alignments for the VIF sequences using the following parameters:</p>
<pre><code>https://www.hiv.lanl.gov/cgi-bin/NEWALIGN/align.cgi

Alignment type: Web alignment (all complete sequences)
Year: 2018
Organism: HIV-1/SIVcpz
DNA/Protein: PRO
Region: vif
Subtype: NO Recombinants (A-K)
Format: FASTA
Alignment ID : 118AP15
Number of sequences: 3412</code></pre>
</div>
<div id="calculate-frequency-of-individual-residues." class="section level1">
<h1>Calculate frequency of individual residues.</h1>
<p>We use the alignments to pull the resodues in positions 31, 33 and 128. We then calculate the frequencies at which the residues occur at a given position in all VIF sequences. Only the three most abundant residues in each position individually listed, the rest are collapsed in the category “other”. The frequencies are represented in the table and bar plot below.</p>
<table class="table" style="width: auto !important; float: left; margin-right: 10px;">
<thead>
<tr>
<th style="text-align:left;">
pos
</th>
<th style="text-align:left;">
res
</th>
<th style="text-align:right;">
freq
</th>
<th style="text-align:right;">
perc
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
31
</td>
<td style="text-align:left;">
I
</td>
<td style="text-align:right;">
1346
</td>
<td style="text-align:right;">
0.620
</td>
</tr>
<tr>
<td style="text-align:left;">
31
</td>
<td style="text-align:left;">
V
</td>
<td style="text-align:right;">
474
</td>
<td style="text-align:right;">
0.218
</td>
</tr>
<tr>
<td style="text-align:left;">
31
</td>
<td style="text-align:left;">
R
</td>
<td style="text-align:right;">
82
</td>
<td style="text-align:right;">
0.038
</td>
</tr>
<tr>
<td style="text-align:left;">
31
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:right;">
269
</td>
<td style="text-align:right;">
0.124
</td>
</tr>
<tr>
<td style="text-align:left;">
33
</td>
<td style="text-align:left;">
K
</td>
<td style="text-align:right;">
1255
</td>
<td style="text-align:right;">
0.578
</td>
</tr>
<tr>
<td style="text-align:left;">
33
</td>
<td style="text-align:left;">
G
</td>
<td style="text-align:right;">
442
</td>
<td style="text-align:right;">
0.204
</td>
</tr>
<tr>
<td style="text-align:left;">
33
</td>
<td style="text-align:left;">
R
</td>
<td style="text-align:right;">
431
</td>
<td style="text-align:right;">
0.199
</td>
</tr>
<tr>
<td style="text-align:left;">
33
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:right;">
43
</td>
<td style="text-align:right;">
0.020
</td>
</tr>
<tr>
<td style="text-align:left;">
128
</td>
<td style="text-align:left;">
I
</td>
<td style="text-align:right;">
1078
</td>
<td style="text-align:right;">
0.497
</td>
</tr>
<tr>
<td style="text-align:left;">
128
</td>
<td style="text-align:left;">
R
</td>
<td style="text-align:right;">
729
</td>
<td style="text-align:right;">
0.336
</td>
</tr>
<tr>
<td style="text-align:left;">
128
</td>
<td style="text-align:left;">
L
</td>
<td style="text-align:right;">
175
</td>
<td style="text-align:right;">
0.081
</td>
</tr>
<tr>
<td style="text-align:left;">
128
</td>
<td style="text-align:left;">
other
</td>
<td style="text-align:right;">
189
</td>
<td style="text-align:right;">
0.087
</td>
</tr>
</tbody>
</table>
<p><img src="B_cluster_files/figure-html/barplot1-1.png" width="480" style="display: block; margin: auto 0 auto auto;" /></p>
<p><br/><br/></p>
</div>
<div id="calculate-frequency-of-pairs-of-residues." class="section level1">
<h1>Calculate frequency of pairs of residues.</h1>
<p>We use the alignments to find which the residues where found at positions 31 and 128 for VIF sequence in each isolate. We then calculate the frequencies at which pairs of residues occur. Note that only selected pairs of amino acids are shown and the rest are collapsed in the category “other”. The frequencies are represented in the table and pie charts below.</p>
<table class="table" style="width: auto !important; float: left; margin-right: 10px;">
<thead>
<tr>
<th style="text-align:left;">
res
</th>
<th style="text-align:right;">
freq
</th>
<th style="text-align:right;">
prec
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
II
</td>
<td style="text-align:right;">
630
</td>
<td style="text-align:right;">
0.290
</td>
</tr>
<tr>
<td style="text-align:left;">
IR
</td>
<td style="text-align:right;">
488
</td>
<td style="text-align:right;">
0.225
</td>
</tr>
<tr>
<td style="text-align:left;">
other
</td>
<td style="text-align:right;">
660
</td>
<td style="text-align:right;">
0.304
</td>
</tr>
<tr>
<td style="text-align:left;">
VI
</td>
<td style="text-align:right;">
267
</td>
<td style="text-align:right;">
0.123
</td>
</tr>
<tr>
<td style="text-align:left;">
VR
</td>
<td style="text-align:right;">
126
</td>
<td style="text-align:right;">
0.058
</td>
</tr>
</tbody>
</table>
<p><img src="B_cluster_files/figure-html/piechart1-1.png" width="480" style="display: block; margin: auto 0 auto auto;" /></p>
</div>
<div id="trees" class="section level1">
<h1>Trees</h1>
<p>We now look at the pylogenetic distribution of residue combinations. In order to assess whether there is anything special about the way VIF proteins relate to each other we will compare them with other proteins, namely NEF, GAG and ENV. Alignments for NEF, GAG and ENV are retrieve from <em>Los Alamos</em> HIV database as decribed in <strong>Data aquisition</strong>. The details of the alignments are found below.</p>
<pre><code>Alignment type: Web alignment (all complete sequences)
Year: 2018
Organism: HIV-1/SIVcpz
DNA/Protein: PRO
Region: nef
Subtype: NO Recombinants (A-K)
Format: FASTA
Alignment ID : 118AP7
Number of sequences: 5340

Alignment type: Web alignment (all complete sequences)
Year: 2018
Organism: HIV-1/SIVcpz
DNA/Protein: PRO
Region: env
Subtype: NO Recombinants (A-K)
Format: FASTA
Alignment ID : 118AP2
Number of sequences: 4760

Alignment type: Web alignment (all complete sequences)
Year: 2018
Organism: HIV-1/SIVcpz
DNA/Protein: PRO
Region: gag
Subtype: NO Recombinants (A-K)
Format: FASTA
Alignment ID : 118AP4
Number of sequences: 6381</code></pre>
<p>To provide an accurate comparison across sequenes, we only take into consideration isolates that have a representative in each protein file. In consequence, all the trees have the same number of sequences.</p>
<p>Trees’ were calculated using the ‘Phylogenetic tree’ functionality from the web server Clustal Omega (<a href="https://www.ebi.ac.uk/Tools/msa/clustalo/" class="uri">https://www.ebi.ac.uk/Tools/msa/clustalo/</a>). Briefly, each file containing sequences for either VIF, ENV, GAG or NEF were uploaded in the server, processed for multiple sequence alignment, the phylogenetic trees were drawn and the phylgenetic tree infromation downloaded as Newick files. There were use as input in ‘ggtree’ for tree display highlighting the position of the isoalted with selected residue combinations.</p>
<p>A total of <strong>795</strong> entries are shared across alignments of Env Gag, Nef and Vif.</p>
<p><img src="B_cluster_files/figure-html/residue_trees-1.png" width="672" /></p>
<pre><code>## R version 3.6.1 (2019-07-05)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.6
## 
## Matrix products: default
## BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8
## 
## attached base packages:
## [1] grid      stats     graphics  grDevices utils     datasets  methods  
## [8] base     
## 
## other attached packages:
##  [1] kableExtra_1.1.0  gridExtra_2.3     viridis_0.5.1    
##  [4] viridisLite_0.3.0 knitr_1.25        dplyr_0.8.3      
##  [7] ggplot2_3.2.1     WriteXLS_5.0.0    seqinr_3.6-1     
## [10] cowplot_1.0.0     ggtree_1.16.6     treeio_1.8.2     
## 
## loaded via a namespace (and not attached):
##  [1] tidyselect_0.2.5   xfun_0.10          purrr_0.3.2       
##  [4] lattice_0.20-38    colorspace_1.4-1   vctrs_0.2.0       
##  [7] htmltools_0.4.0    yaml_2.2.0         rlang_0.4.0       
## [10] pillar_1.4.2       glue_1.3.1         withr_2.1.2       
## [13] rvcheck_0.1.5      lifecycle_0.1.0    stringr_1.4.0     
## [16] munsell_0.5.0      gtable_0.3.0       rvest_0.3.4       
## [19] evaluate_0.14      labeling_0.3       parallel_3.6.1    
## [22] highr_0.8          Rcpp_1.0.2         readr_1.3.1       
## [25] scales_1.0.0       backports_1.1.5    BiocManager_1.30.7
## [28] webshot_0.5.1      jsonlite_1.6       hms_0.5.1         
## [31] digest_0.6.21      stringi_1.4.3      ade4_1.7-13       
## [34] tools_3.6.1        magrittr_1.5       lazyeval_0.2.2    
## [37] tibble_2.1.3       crayon_1.3.4       ape_5.3           
## [40] tidyr_1.0.0        pkgconfig_2.0.3    zeallot_0.1.0     
## [43] MASS_7.3-51.4      tidytree_0.2.8     xml2_1.2.2        
## [46] httr_1.4.1         assertthat_0.2.1   rmarkdown_1.16    
## [49] rstudioapi_0.10    R6_2.4.0           nlme_3.1-141      
## [52] compiler_3.6.1</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
